packages/react-strict-dom/src/native/modules/createStrictDOMComponent.js

```diff
@@ -676,14 +679,16 @@ export function createStrictDOMComponent<T, P: StrictProps>(
         );
       }
 
-      const inheritedCustomProperties = React.useContext(ThemeContext);
-
       return (
         <ThemeContext.Provider
-          value={{
-            ...inheritedCustomProperties,
-            ...customPropertiesFromThemes
-          }}
+          value={
+            customPropertiesFromThemes != null
+              ? {
+                  ...inheritedCustomProperties,
+                  ...customPropertiesFromThemes
+                }
+              : inheritedCustomProperties
```

Comment:
Avoid creating a new object if we don't have any new custom properties passed in via the `style` prop for this element. In another diff we can skip the provider wrapper too, but trying to limit this change to just object copies for the purposes of assessing the hypothesis we had from profiling.